src/servenix/server/servenix.py

```diff
@@ -55,6 +65,21 @@ def __init__(self, nix_store_path, nix_bin_path, nix_state_path,
             self._nar_extension = ".nar.xz"
         # Enable interactive debugging on unknown errors.
         self._debug = debug
+        # Test connect to the nix database; if successful, then we
+        # will use a direct connection to the database rather than
+        # using nix-store. This is much faster, but is unavailable on
+        # some systems.
+        try:
+            db_path = join(nix_state_path, "nix", "db", "db.sqlite")
+            query = "select * from ValidPaths limit 1"
+            db_con = sqlite3.connect(db_path)
+            db_con.execute(query).fetchall()
+            # If this succeeds, assign the db_con attribute.
+            self._db_con = db_con
+        except Exception as err:
```

Comment:
I considered that but I'm worried about it being too specific; essentially if any step within that try block fails then we can't connect to the database and we want to bomb out.